plugins/outputs/amqp/amqp.go

```diff
@@ -210,9 +210,23 @@ func (q *AMQP) Close() error {
 
 func (q *AMQP) routingKey(metric telegraf.Metric) string {
 	if q.RoutingTag != "" {
-		key, ok := metric.GetTag(q.RoutingTag)
-		if ok {
-			return key
+		var routingTags = strings.Split(q.RoutingTag, ".")
+		var validTags = make([]string, 0, len(routingTags))
+		k := 0
+		for _, tag := range routingTags {
+			if tag != "" {
+				key, ok := metric.GetTag(tag)
+				if ok {
+					// extend the slice for the new key
+					validTags = validTags[0 : len(validTags)+1]
+					validTags[k] = key
```

Comment:
Change this to `validTags = append(validTags, key)`, don't change line 214, but remove `k`s initialization.